### DISCUSSION TITLE
BT and ET negative

### DISCUSSION THREAD

**Discussion by ProseCoffee** [artisan-roaster-scope/artisan#1245] [General]
Hello, 
I have zero knowledge of computer programming, and while I love artisan when it works, it can be a beast for me to fix when it doesn't. I am currently having an issue where I upgraded to the newest version, and now I can't get the BT and ET to register. Depending on what settings I have, it either registers u.u for BT and ET, or it registers as -0.0 for both. I have checked all of the MODBUS settings that I can find, have updated my serialcomm drivers, and have spent hours scouring the forums for other ideas. I am now a roasting session behind schedule because I spent the last one trying to fix my program. 
As a side note, since the most recent update, I am also having the crashing issue but it is intermittent, usually when I turn on and reset. I run a Mac OS 13.4.1. I haven't been able to do a roast since updating because I have had the temp issues, but I'm hoping it doesn't crash during a roast. 
I appreciate any help with this! 

Sharon

**Comment by roasterdave** [comment]
What are you connecting to via MODBUS?  Can you tell us what serialcomm drivers you upgraded?  

Upgrading Artisan does not change any settings.  If you saved settings to a file before the upgrade we can try restoring them (Help>> Load settings).  If not we'll start at the beginning.

**Reply by ProseCoffee** [reply]
Hi Dave, 
I went through the San Fran website and downloaded the drivers that were recommended for the sf6 on a Mac. It's listed as Isolated USB To R$485/RS422 Adapter 2.0 FTDI Drivers, they're in beta but I'm pretty sure that it's the same one I used before. I just did a factory reset on artisan to see if that would help, but it it's still doing the same thing.

**Reply by roasterdave** [reply]
This warning is on the [Artisan-Scope.org](https://artisan-scope.org/modbus_serial/) site.  
`Watch out! The FTDI driver is preinstalled on virtually all Linux distributions as well as all OS X versions supported by Artisan. Installing an additional FTDI driver on those operating system might lead to instabilities!`

That could  be the problem.  This [article](https://ftdichip.com/Support/Knowledgebase/index.html?macosxuninstallingvcp.htm) may help if you choose to uninstall the additional driver.   I don't use a Mac so my ability to help with the driver is limited.  

Before going down that path, have you tried going through the machine setup again?  Config>> Machine>> San Franciscan then choose the appropriate controller, either SF(Watlow) or Eurotherm.

**Reply by ProseCoffee** [reply]
I had seen the warning on the artisan-scope website, and tried uninstalling the drivers that I had downloaded, but that didn't work either. It also doesn't make a ton of sense because it worked before with the installed drivers. 
I have tried going through the machine setup again, it is still giving me the same connection error.
I appreciate your help, I know nothing about these things!

**Reply by roasterdave** [reply]
Does your SF have a Watlow or a Eurotherm controller?  Can you confirm you chose the appropriate one in the machine setup?  The Watlow is the plain SF setup.

**Comment by ProseCoffee** [comment]
I just tried restoring both of the previous save files that I have and neither of them worked, so from the beginning it is!
